Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabSaveLoadHandler.h

```diff
@@ -60,6 +68,7 @@ namespace AzToolsFramework
             : public QObject
             , public AzQtComponents::DragAndDropEventsBus::Handler
             , public AzQtComponents::DragAndDropItemViewEventsBus::Handler
+            , private AzToolsFramework::AssetSystemBus::Handler
```

Comment:
The PrefabSaveHandler class seems specific to Qt/UI events. Would it make more sense to move the prefab asset change handling to a different class?

Reply:
OK. Perhaps we can change the comment describing the PrefabSaveLoadHandler class for now then.

Reply:
I think the PrefabSaveLoadHandler file needs to be refactored a bit because the file name is PrefabSaveLoadHandler but the class name is PrefabSaveHandler. And its also not clear whats the difference between PrefabSaveLoadHandler and PrefabLoader. I'll do the refactoring as part of another PR

Reply:
It would be good to something else rather than `PrefabSaveLoadHandler` since it is handling multiple functionalities except save/load. Currently, it also handles drag-and-drop logic. Alternatively, we should break it down into different classes before it gets assigned more responsibility. 😀